gateway/1.5/modules/ROOT/pages/policies-pdk-configure-features.adoc

```diff
@@ -1,5 +1,589 @@
+----
+
+Instead of filtering an incoming request, you can define a function that processes the outgoing response by using the `on_response` wrapper instead of the `on_request` wrapper:
+
+[source,Rust]
+----
+async fn response_filter(response_state: ResponseState, _config: &Config) {
+    ...
+}
+
+#[entrypoint]
+async fn configure(launcher: Launcher, Configuration(bytes): Configuration) -> Result<()> {
+    let config = serde_json::from_slice(&bytes)?;
+    let filter = on_response(|rs| response_filter(rs, &config));
+    launcher.launch(filter).await?;
+    Ok(())
+}
+----
+
+NOTE: The `request_filter` function is now the `response_filter` function.
+
+To filter both requests and response, use bother wrappers:
+
+[source,Rust]
+----
+async fn request_filter(request_state: RequestState, _config: &Config) {
+   ...
+}
+
+async fn response_filter(response_state: ResponseState, _config: &Config) {
+    ...
+}
+
+#[entrypoint]
+async fn configure(launcher: Launcher, Configuration(bytes): Configuration) -> Result<()> {
+    let config = serde_json::from_slice(&bytes)?;
+    let filter = on_request(|rs| request_filter(rs, &config))
+        .on_response(|rs| response_filter(rs, &config));
+    launcher.launch(filter).await?;
+    Ok(())
+}
+----
+
+The `on_response` or `on_request` wrapper defines when the request is executed. 
+
+You can define functions other than the provided `requests_filter` and `response_filter` and define them from the `#[entrypoint]` function. The `requests_filter` and `response_filter`` are only examples.
+
+== Configure Policy Logging
+
+PDK provides a logging mechanism that generates a log message enriched with the API instance ID, policy ID, and request ID.
+
+Insert custom logs with the following macros by using the `logger::` package:
+
+* `logger::debug!`
+* `logger::info!`
+* `logger::warn!`
+* `logger::error!`
+
+The macros behave the same as the https://doc.rust-lang.org/std/macro.format.html[Rust std::format! macro].  The first parameter must be a format string literal. Use `{}` in the literal to pass parameters, for example:
+
+[source,Rust]
+----
+let value = "there!";
+logger::debug!("Hello there!");
+logger::info!("Hello {}", value);
+logger::warn!("Hello {value}");
+logger::error!("Hello {}", "there!");    into_headers_state().await
+----
+
+All examples appear in the Flex Gateway logs in the following format:
+----
+[flex-gateway-envoy][<log-level>] wasm log <policy-name>.<api-instance-name> main: [policy: <policy-name>][api: <api-instance-name>][req: <request-id> ] Hello there!
+----
+
+For more information about viewing Flex Gateway logs, see xref:flex-gateway-monitor.adoc[].
+
+== Read and Write Headers
+
+To access the headers in both the request and the response, transform the `RequestState` or `ResponseState` to a header state by calling the method `into_headers_state` and awaiting it. After calling either method, access and manipulate the headers by calling the functions of the `HeadersHandler` trait.
```

Comment:
"and awaiting it." 
Not sure what this means, do you wait for the method to complete? Do users have to set some kind of listener that waits until the method finishes? Can't we just remove this last part and end the sentence with "by calling the method `into_headers_state`"? Please confirm.

Reply:
I'm not exactly sure how it works but since Rust is a multi-thread language there's a .operator calles "await" that signifies dependencies. This is the function used in the code below "request_state.into_headers_state().await".

Sometimes you need to await functions but sometimes you do not.